CLIENTS.md

```diff
@@ -0,0 +1,264 @@
+
+Rekor v2 supports HTTP or gRPC, like Fulcio. For Go, we have implemented a client
+already. For other languages, they can either use the OpenAPI docs or generated
+gRPC service proto, or create their own client.
+
+The service implements one write API, `/api/v2/log/entries`. Example JSON request bodies
+below:
+
+```json
+// Request with a Fulcio certificate
+{
+    "hashedRekordRequestV0_0_2": {
+        "digest": "<base64 digest of artifact>", # Must use hash algorithm from key_details
+        "signature": {
+            "content": "<base64 signature>",
+            "verifier": {
+                "x509Certificate": "<base64 DER-encoded certificate>",
+                "keyDetails": "PKIX_ECDSA_P256_SHA_256" # Must match signing algorithm
+            }
+        }
+    }
+}
+
+// Request with a self-managed key
+{
+    "hashedRekordRequestV0_0_2": {
+        "digest": "<base64 digest of artifact>",
+        "signature": {
+            "content": "<base64 signature>",
+            "verifier": {
+                "publicKey": {
+                    "raw_bytes": "<base64 DER-encoded public key>"
+                },
+                "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+            }
+        }
+    }
+}
+
+// Request with an attestation
+{
+    "dsseRequestV0_0_2": {
+        "envelope": {
+            "payload": "<base64-encoded message>",
+            "payloadType": "<type, e.g. application/vnd.in-toto+json>",
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "x509Certificate": "<base64 DER-encoded certificate>", # May also be a public key
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256" # Must match signing algorithm
+        }
+    }
+}
+```
+
+The response will be a `TransparencyLogEntry` message, which should be persisted
+in a bundle. Clients no longer need to transform the Rekor response into a `TLE`
+message to store in the bundle.
+
+## Handling Longer Requests
+
+Clients need to increase request timeouts to at least 10 seconds.
+
+Rekor now batches uploads so that checkpoints are published less frequently.
+Additionally, we plan to support synchronous witnessing, where third-party
+witnesses independently verify the consistency of the log and Rekor provides
+co-signed checkpoints with each upload response.
+
+## Signed RFC 3161 Timestamps
+
+Rekor will no longer return SignedEntryTimestamps or include integrated time
+in the response. Clients must fetch an RFC 3161 signed timestamp from a trusted
+timestamp authority and include the signed timestamp in the bundle.
+
+Sigstore now operates a timestamp authority at `timestamp.sigstore.dev` and
+`timestamp.sigstage.dev` for staging, and the roots for these services will
+be included in the TrustedRoot distributed via TUF. Clients may request timestamps
+from other trusted timestamp authorities as well.
```

Comment:
I assume we are expecting clients that use other "trusted timestamp" authorities to inject that information into the trusted root? Client code should not fetch the timestamp cert from the provider